src/overlays/actors/ovl_En_Ishi/z_en_ishi.c

```diff
@@ -21,47 +55,420 @@ const ActorInit En_Ishi_InitVars = {
+        fragmentVelocity.y += (Math_Rand_ZeroOne() * 4.0f) + ((Math_Rand_ZeroOne() * i) * 0.7f);
+        fragmentVelocity.z += rand * Math_Coss(angle);
+        if (i == 0) {
+            phi_v0 = 41;
+            phi_v1 = -450;
+        } else if (i < 4) {
+            phi_v0 = 37;
+            phi_v1 = -380;
+        } else {
+            phi_v0 = 69;
+            phi_v1 = -320;
+        }
+        EffectSsKakera_Spawn(globalCtx, &effectPos, &fragmentVelocity, &this->actor.posRot.pos, phi_v1, phi_v0, 30, 5,
+                             0, scales[i], 5, 2, 70, KAKERA_COLOR_WHITE, OBJECT_GAMEPLAY_FIELD_KEEP, D_0500A5E8);
+    }
+}
+
+void EnIshi_SpawnDustSmall(EnIshi* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+
+    Math_Vec3f_Copy(&pos, &this->actor.posRot.pos);
+    if (this->actor.bgCheckFlags & 1) {
+        pos.x += 2.0f * this->actor.velocity.x;
+        pos.y -= 2.0f * this->actor.velocity.y;
+        pos.z += 2.0f * this->actor.velocity.z;
+    } else if (this->actor.bgCheckFlags & 8) {
+        pos.x -= 2.0f * this->actor.velocity.x;
+        pos.y += 2.0f * this->actor.velocity.y;
+        pos.z -= 2.0f * this->actor.velocity.z;
+    }
+    func_80033480(globalCtx, &pos, 60.0f, 3, 0x50, 0x3C, 1);
+}
+
+void EnIshi_SpawnDustLarge(EnIshi* this, GlobalContext* globalCtx) {
+    Vec3f pos;
+
+    Math_Vec3f_Copy(&pos, &this->actor.posRot.pos);
+    if (this->actor.bgCheckFlags & 1) {
+        pos.x += 2.0f * this->actor.velocity.x;
+        pos.y -= 2.0f * this->actor.velocity.y;
+        pos.z += 2.0f * this->actor.velocity.z;
+    } else if (this->actor.bgCheckFlags & 8) {
+        pos.x -= 2.0f * this->actor.velocity.x;
+        pos.y += 2.0f * this->actor.velocity.y;
+        pos.z -= 2.0f * this->actor.velocity.z;
+    }
+    func_80033480(globalCtx, &pos, 140.0f, 0xA, 0xB4, 0x5A, 1);
+}
+
+void EnIshi_DropCollectible(EnIshi* this, GlobalContext* globalCtx) {
+    if (!(this->actor.params & 1)) {
+        s16 dropParams = (this->actor.params >> 8) & 0xF;
+        if (dropParams >= 0xD) {
+            dropParams = 0;
+        }
+        Item_DropCollectibleRandom(globalCtx, NULL, &this->actor.posRot.pos, dropParams << 4);
+    }
+}
+
+void EnIshi_Fall(EnIshi* this) {
+    this->actor.velocity.y += this->actor.gravity;
+
+    if (this->actor.velocity.y < this->actor.minVelocityY) {
+        this->actor.velocity.y = this->actor.minVelocityY;
+    }
+}
+
+void func_80A7ED94(Vec3f* arg0, f32 arg1) {
+    arg1 += ((Math_Rand_ZeroOne() * 0.2f) - 0.1f) * arg1;
+    arg0->x -= arg0->x * arg1;
+    arg0->y -= arg0->y * arg1;
+    arg0->z -= arg0->z * arg1;
+}
+
+void EnIshi_SpawnBugs(EnIshi* this, GlobalContext* globalCtx) {
+    s32 i;
+
+    for (i = 0; i < 3; i++) {
+        if (Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_INSECT, this->actor.posRot.pos.x,
+                        this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, Math_Rand_ZeroOne() * 0xFFFF, 0,
+                        1) == NULL) {
```

Comment:
if you can afford a temp actor pointer for bugs I think this would read alot nicer. just imo tho, and only if it works of course. itd be like
```
bug = Actor_Spawn();
if (bug == NULL) {
    break;
}
```